tests/test_utilblocks.py

```diff
@@ -0,0 +1,70 @@
+# Standard
+from unittest.mock import MagicMock, patch
+import unittest
+
+# Third Party
+from datasets import Dataset, Features, Value
+
+# First Party
+from src.instructlab.sdg.utilblocks import (
+    DuplicateColumns,
+    FlattenColumnsBlock,
+    RenameColumns,
+    SetToMajorityValue,
+)
+
+
+class TestUtilBlock(unittest.TestCase):
+    def setUp(self):
+        self.mock_ctx = MagicMock()
+        self.mock_pipe = MagicMock()
+
+        self.dataset = Dataset.from_dict(
+            {
+                "name": ["Tomato", "Rose", "Carrot", "Basil"],
+                "type": ["Fruit", "Flower", "Vegetable", "Herb"],
+                "edible": ["yes", "no", "yes", "yes"],
+            }
+        )
+
+    def test_dup(self):
+        dup_block = DuplicateColumns(
+            self.mock_ctx, self.mock_pipe, "blockname", columns_map={"edible": "caneat"}
+        )
+        assert "caneat" not in self.dataset.column_names
+        new_samples = dup_block.generate(self.dataset)
+        assert "edible" in new_samples.column_names
+        assert "caneat" in new_samples.column_names
+
+    def test_renamecolumns(self):
+        rename_block = RenameColumns(
+            self.mock_ctx, self.mock_pipe, "blockname", columns_map={"edible": "caneat"}
+        )
+        new_samples = rename_block.generate(self.dataset)
+        assert "caneat" in new_samples.column_names
+        assert "edible" not in new_samples.column_names
+
+    def test_setmajority(self):
+        mv_block = SetToMajorityValue(
+            self.mock_ctx, self.mock_pipe, "blockname", col_name="edible"
+        )
+
+        new_samples = mv_block.generate(self.dataset)
+
+        for row in new_samples.select_columns(["edible"]):
+            assert row.get("edible") == "yes"
```

Comment:
I don't really understand this one. Does the block need to support strings, or is it really aimed at numeric values only? How should someone expect it to behave on strings if we need to keep string support?

Reply:
I don't think its confined to numeric values (although I haven't see it used) , it selects the most common value and sets all column values to match it